app/vep/models/upload_vcf_files.py

```diff
@@ -0,0 +1,49 @@
+import logging
+import tempfile
+import aiofiles
+
+import typing
+from starlette import status
+
+from fastapi import UploadFile, File, HTTPException
+
+from core.config import VCF_UPLOAD_DIRECTORY
+
+ALLOWED_FILE_TYPES = ("vcf", "vcf.gz", ".txt")  # Allowed file types
+MAX_FILE_SIZE = 240 * 1024 * 1024  # 250 MB
+
+
+async def validate_file(input_file: UploadFile = File(...), allowed_types: typing.Tuple[str] = ALLOWED_FILE_TYPES,
+                        max_size: int = MAX_FILE_SIZE) -> bool:
+    # Check file type
+    if not input_file.filename.endswith(ALLOWED_FILE_TYPES):
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            detail=f"Invalid file type: {input_file.content_type}. Allowed types are: {', '.join(allowed_types)}."
+        )
```

Comment:
The VEP tool should do input format checking on its own, regardless of extension (see Nakib's comment on [Slack](https://genomes-ebi.slack.com/archives/C032E9YJ2N9/p1720433684433359)). I don't think we should reject files based on file extensions on our side.